src/main/java/org/folio/rest/impl/EholdingsPackagesImpl.java

```diff
@@ -49,25 +54,70 @@
 
 
   private RMAPIConfigurationService configurationService;
+  private PackagesConverter converter;
   private HeaderValidator headerValidator;
+  private PackageParametersValidator packageParametersValidator;
 
 
   public EholdingsPackagesImpl() {
     this(
       new RMAPIConfigurationServiceCache(
         new RMAPIConfigurationServiceImpl(new ConfigurationClientProvider())),
-      new HeaderValidator());
+      new HeaderValidator(),
+      new PackageParametersValidator(),
+      new PackagesConverter());
   }
 
   public EholdingsPackagesImpl(RMAPIConfigurationService configurationService,
-                               HeaderValidator headerValidator) {
+                               HeaderValidator headerValidator, PackageParametersValidator packageParametersValidator,
+    PackagesConverter converter) {
     this.configurationService = configurationService;
     this.headerValidator = headerValidator;
+    this.packageParametersValidator = packageParametersValidator;
+    this.converter = converter;
   }
 
   @Override
-  public void getEholdingsPackages(String filterCustom, String q, String filterSelected, String filterType, String sort, int page, int count, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(Future.succeededFuture(Response.status(Response.Status.NOT_IMPLEMENTED).build()));
+  @Validate
+  @HandleValidationErrors
+  public void getEholdingsPackages(String filterCustom, String q, String filterSelected,
+    String filterType, String sort, int page, int count, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    headerValidator.validate(okapiHeaders);
+    packageParametersValidator.validate(filterCustom, filterSelected, filterType, sort);
+
+    boolean isFilterCustom = Boolean.parseBoolean(filterCustom);
+    Sort nameSort = sort != null ? Sort.valueOf(sort.toUpperCase()) : null;
```

Comment:
sortable.raml was changed to have "default: relevance", so now sort is guaranteed not to be null

Reply:
thanks Andrii, will change the code